src/core/server/http/base_path_service.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Request } from 'hapi';
+import { KibanaRequest } from './router';
+
+import { modifyUrl } from '../../utils';
+
+export class BasePath {
+  private readonly basePathCache = new WeakMap<
+    ReturnType<KibanaRequest['unstable_getIncomingMessage']>,
+    string
+  >();
+
+  constructor(private readonly basePath?: string) {}
```

Comment:
It might be more clear if this member is called `serverBasePath` rather than just `basePath`.